models/utils/cohort_id_format.py

```diff
@@ -0,0 +1,81 @@
+from typing import Iterable
+
+from api.settings import COHORT_CHECKSUM_OFFSET, COHORT_PREFIX
+from models.utils.luhn import luhn_compute, luhn_is_valid
+
+
+def cohort_id_format(cohort_id: int | str) -> str:
+    """
+    Transform raw (int) cohort identifier to format (COHXXX) where:
+        - COH is the prefix
+        - XXX is the original identifier
+        - H is the Luhn checksum
+    """
+    # Validate input
+    if isinstance(cohort_id, str) and not cohort_id.isdigit():
+        if cohort_id.startswith(COHORT_PREFIX):
+            return cohort_id
+        raise ValueError(f'Unexpected format for cohort identifier {cohort_id!r}')
+
+    cohort_id = int(cohort_id)
+
+    checksum = luhn_compute(
+        cohort_id, offset=COHORT_CHECKSUM_OFFSET
+    )
+
+    return f'{COHORT_PREFIX}{cohort_id}{checksum}'
+
+
+def cohort_id_format_list(cohort_ids: Iterable[int | str]) -> list[str]:
+    """
+    Transform LIST of raw (int) cohort identifier to format (COHXXX) where:
+        - COH is the prefix
+        - XXX is the original identifier
+        - H is the Luhn checksum
+    """
+    return [cohort_id_format(s) for s in cohort_ids]
+
+
+def cohort_id_transform_to_raw(cohort_id: int | str) -> int:
+    """
+    Transform STRING cohort identifier (COHXXXH) to XXX by:
+        - validating prefix
+        - validating checksum
+    """
+    expected_type = str
+    if not isinstance(cohort_id, expected_type):  # type: ignore
+        raise TypeError(
+            f'Expected identifier type to be {expected_type!r}, received {type(cohort_id)!r}'
+        )
+
+    if isinstance(cohort_id, int):
+        return cohort_id
+
+    if not isinstance(cohort_id, str):
+        raise ValueError('Programming error related to cohort checks')
```

Comment:
MH: Isn't this line redundant? LN 46 should take care of this.

Reply:
Yeah, as per above, I've cp'd this entire script. But I agree with you and have removed the line!